### ISSUE TITLE
2b2t gives error "you have lost connection to the server" or "invalid session"

### ISSUE THREAD

**Issue by waterdragon78** [themoonisacheese/2bored2wait#100]
When I use it 2b2w and it does launch, 2b2t just says "you have lost connection to the server" and "Invalid session"
Can someone please help me? If it helps, I'm on macOS 10.14. Port 8080 for the webserver. Also, why does it show queue priority in the terminal window when it says you get into the server in 5 seconds?

**Comment by waterdragon78** [comment]
please I really need help

**Comment by MrGeorgen** [comment]
Did connect to the proxy and not 2b2t?

**Comment by waterdragon78** [comment]
If by proxy u mean connected to 2b2w then yes I’m connected, but when I try to connect to 2b2t with the proxy on, the error happens

**Comment by waterdragon78** [comment]
Hello?

**Comment by MrGeorgen** [comment]
If you to 2b2t directly from your minecraft client , 2b2w gets kicked out. 2b2w reconnects automatically, so can't connect. Make sure you connect to 2b2w with correct port set in config.json (`localhost:25566` by default)

**Comment by waterdragon78** [comment]
Hello?

**Comment by waterdragon78** [comment]
So I do the correct port for the server, does the port have to be the same for the web server?

**Comment by waterdragon78** [comment]
hewo?

**Comment by themoonisacheese** [comment]
the port has to specifically be different, otherwise you have 2 things on the same port and that's what's causing the problems.

**Comment by waterdragon78** [comment]
But let’s say the queue finished, then do I go into Minecraft? And no queue?

**Comment by dsetareh** [comment]
You need to connect to the instance of 2b2w that you are running. It is acting as a proxy/bouncer for your 2b2t connection.

**Comment by waterdragon78** [comment]
Can u simplify it a bit?

**Comment by waterdragon78** [comment]
So I just connect when the queue is done?

**Comment by dsetareh** [comment]
ELI5:

2bored2wait pretends to be you and logs into 2b2t's queue

while 2bored2wait is pretending to be you in queue, you cannot login to any other servers because 2bored2wait would get kicked off

instead of connecting to 2b2t and joining queue, you connect to 2bored2wait which is relaying data to you

Think of it as if you're using a minecraft VPN to connect to 2b2t.

**Comment by waterdragon78** [comment]
so how do I get onto the server?

**Comment by themoonisacheese** [comment]
have you watched the explanation video at all?

**Comment by waterdragon78** [comment]
THERE'S A VIDEO???!?!?!?!?!?

**Comment by themoonisacheese** [comment]
![image](https://user-images.githubusercontent.com/15526977/94562398-9504bc00-0265-11eb-88f0-a886c842ab75.png)

**Comment by waterdragon78** [comment]
Oh geez I'm so sorry I didn't see that

**Comment by waterdragon78** [comment]
Sorry
